enzyme/Enzyme/ActivityAnalysis.cpp

```diff
@@ -1136,7 +1187,7 @@ bool ActivityAnalyzer::isConstantValue(TypeResults &TR, Value *Val) {
 
           // If requesting empty unknown functions to be considered inactive,
           // abide by those rules
```

Comment:
This no longer appears to be correct since a cacheable function doesn't necessarily mean it is (in)active and vice versa.

Reply:
Rename from `isCertainCacheable` to `isCertainInactiveAndCacheable`, or do you have another suggestion?
At least with the function I added here I think it currently still seems to hold and we can add another function / logic later, once we consider more functions to cover all 2*2 combinations?

Reply:
I mean this location solely cares about activity, nothing about whether or not it is cacheable.

Regardless, here it should be safe to just change this to:

```
          if (EnzymeEmptyFnInactive && called->empty() &&
              !hasMetadata(called, "enzyme_gradient") &&
              !hasMetadata(called, "enzyme_derivative") &&
              !isAllocationFunction(called) && !isDeallocationFunction(called) &&
              !isa<IntrinsicInst>(op)) {
 ```
 
 In essence what this code is saying is if one chooses to enable the "all empty functions are inactive by assumption" flag, we still should mark allocation/free functions as being potentially able to create active memory.

Reply:
Most uses of the function are actually as the name implies, a check whether it is a free, allocation, or print, and not a question of cacheability.

Reply:
Updated this if statement and undone the renaming on other locations.

Reply:
I think (?) you need to mark this change request as solved or directly merge this PR. Github won't let me do that.